projection/account/account.go

```diff
@@ -90,9 +103,97 @@ func (projection *Account) HandleEvents(height int64, events []event_entity.Even
+			}
+			if handleErr := projection.addAccountEvent(
+				accountsView,
+				coinSpentEvent.Address,
+				coinSpentEvent,
+				coinSpentEvent.EventName,
+				coinSpentEvent.BlockHeight,
+			); handleErr != nil {
+				return fmt.Errorf("error handling CoinSpentEvent: %v", handleErr)
+			}
+		} else if coinReceivedEvent, ok := event.(*event_usecase.CoinReceived); ok {
+			if handleErr := projection.incrementUsableBalance(accountsView, coinReceivedEvent.Address, coinReceivedEvent.Amount); handleErr != nil {
+				return fmt.Errorf("error handling CoinReceivedEvent: %v", handleErr)
+			}
+
+			if handleErr := projection.addAccountEvent(
+				accountsView,
+				coinReceivedEvent.Address,
+				coinReceivedEvent,
+				coinReceivedEvent.EventName,
+				coinReceivedEvent.BlockHeight,
+			); handleErr != nil {
+				return fmt.Errorf("error handling CoinReceivedEvent: %v", handleErr)
+			}
+		} else if coinMintEvent, ok := event.(*event_usecase.CoinMint); ok {
+			if handleErr := projection.incrementUsableBalance(accountsView, coinMintEvent.Address, coinMintEvent.Amount); handleErr != nil {
+				return fmt.Errorf("error handling coinMintEvent: %v", handleErr)
+			}
+			if handleErr := projection.addAccountEvent(
+				accountsView,
+				coinMintEvent.Address,
+				coinMintEvent,
+				coinMintEvent.EventName,
+				coinMintEvent.BlockHeight,
+			); handleErr != nil {
+				return fmt.Errorf("error handling coinMintEvent: %v", handleErr)
+			}
+		} else if coinBurnEvent, ok := event.(*event_usecase.CoinBurn); ok {
+			if handleErr := projection.decrementUsableBalance(accountsView, coinBurnEvent.Address, coinBurnEvent.Amount); handleErr != nil {
+				return fmt.Errorf("error handling coinBurnEvent: %v", handleErr)
+			}
+			if handleErr := projection.addAccountEvent(
+				accountsView,
+				coinBurnEvent.Address,
+				coinBurnEvent,
+				coinBurnEvent.EventName,
+				coinBurnEvent.BlockHeight,
+			); handleErr != nil {
+				return fmt.Errorf("error handling coinBurnEvent: %v", handleErr)
+			}
+		} else if transactionFailedEvent, ok := event.(*event_usecase.TransactionFailed); ok {
+			feePayer := ""
+			if transactionFailedEvent.FeeGranter != "" {
+				// if set, the fee payer (either the first signer or the value of the payer field) requests that a fee grant be used
+				// to pay fees instead of the fee payer's own balance. If an appropriate fee grant does not exist or the chain does
+				// not support fee grants, this will fail
+				feePayer = transactionFailedEvent.FeeGranter
+			} else if transactionFailedEvent.FeePayer != "" {
+				// if unset, the first signer is responsible for paying the fees. If set, the specified account must pay the fees.
+				// the payer must be a tx signer (and thus have signed this field in AuthInfo).
+				feePayer = transactionFailedEvent.FeePayer
+			} else {
+				signers := projection.ParseSenderAddresses(transactionFailedEvent.Signers)
+				feePayer = signers[0]
+			}
```

Comment:
Just curious, is there any reference on how the fee is paid in CosmosSDK? Thanks!

Reply:
Yea, you may refer to the comments on this issue #383